test/AutoRest.TestServerLowLevel.Tests/dpg-customization.cs

```diff
@@ -60,6 +60,57 @@ await foreach (Page<BinaryData> page in allPages.AsPages())
             }
         });
 
+        [Test]
+        public Task HandwrittenModelLro() => Test(async (host) =>
+        {
+            using var diagnosticListener = new ClientDiagnosticListener("dpg_customization_LowLevel", asyncLocal: true);
+            CollectionAssert.IsEmpty(diagnosticListener.Scopes);
+
+            var lro = await new DPGClient(Key, host).LroValueAsync(WaitUntil.Started, "model");
+            Assert.AreEqual(1, diagnosticListener.Scopes.Count);
+            Assert.AreEqual("DPGClient.LroValue", diagnosticListener.Scopes[0].Name);
+
+            await lro.WaitForCompletionAsync();
+            Assert.AreEqual(2, diagnosticListener.Scopes.Count);
+            Assert.AreEqual("model", $"{lro.Value.Received}");
+            Assert.AreEqual("DPGClient.LroValue.UpdateStatus", diagnosticListener.Scopes[1].Name);
+        });
+
+        [Test]
+        public Task HandwrittenModelLro_ManualIteration() => Test(async (host) =>
+        {
+            using var diagnosticListener = new ClientDiagnosticListener("dpg_customization_LowLevel", asyncLocal: true);
+            CollectionAssert.IsEmpty(diagnosticListener.Scopes);
+
+            var lro = await new DPGClient(Key, host).LroValueAsync(WaitUntil.Started, "model");
+            Assert.AreEqual(1, diagnosticListener.Scopes.Count);
+            Assert.AreEqual("DPGClient.LroValue", diagnosticListener.Scopes[0].Name);
+
+            var updatesCount = 0;
+            while (!lro.HasCompleted)
+            {
+                await lro.UpdateStatusAsync();
+                updatesCount++;
+                Assert.AreEqual("DPGClient.LroValue.UpdateStatus", diagnosticListener.Scopes[updatesCount].Name);
+            }
+
+            // +1 due to the first scope being created by LroValueAsync
+            Assert.AreEqual(updatesCount + 1, diagnosticListener.Scopes.Count);
```

Comment:
We should validate that these are correctly nested/not, not just the count.

Reply:
`ClientDiagnosticListener` doesn't have this functionality out of the box. The only thing I can do is validate that all scopes are closed before next one is opened